sim/druid/runes.go

```diff
@@ -31,6 +32,127 @@ func (druid *Druid) applyFuryOfStormRage() {
 	})
 }
 
+func (druid *Druid) applyEclipse() {
+
+	if !druid.HasRune(proto.DruidRune_RuneBeltEclipse) {
+		return
+	}
+	WrathRanks := len(druid.Wrath)
+	StarfireRanks := len(druid.Starfire)
+
+	// Solar
+	solarProcMultiplier := 30.0
+	druid.SolarEclipseProcAura = druid.RegisterAura(core.Aura{
+		Label:     "Solar Eclipse proc",
+		Duration:  time.Second * 15,
+		MaxStacks: 4,
+		ActionID:  core.ActionID{SpellID: 408250},
+		OnGain: func(aura *core.Aura, sim *core.Simulation) {
+			for rank := 1; rank <= WrathRanks; rank++ {
+				druid.Wrath[rank].BonusCritRating += solarProcMultiplier
+			}
+		},
+		OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+			for rank := 1; rank <= WrathRanks; rank++ {
+				druid.Wrath[rank].BonusCritRating -= solarProcMultiplier
+			}
+		},
+		OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			MatchWrath := false
+
+			// Are we casting Wrath
+			for rank := 1; rank <= WrathRanks; rank++ {
+				if druid.Wrath[rank].IsEqual(spell) {
+					MatchWrath = true
+				}
+			}
```

Comment:
🔨 What you can do here is create a `SpellCode` (see [here](https://github.com/wowsims/sod/blob/bfd146285c13bb9c591899eec31d0e9fe318cae6/sim/shaman/shaman.go#L76-L89)) and apply it to Wrath and Starfire in their respective files where they're registered, then you can do this:
```
if !spell.SpellCode != druid.SpellCode_Wrath {
 return 
}
```